### PR TITLE
Alerting users attempting to put a date in event title

### LINKED ISSUE #891: Prevent Years in Event Titles
### User story
As an administrator reviewing requests, sometimes people include years in the event name and sometimes they don't. This causes the requests for the same event to be split between two or more groups, making it difficult to review them together (or requiring work to rename events). I would like users to be warned before including years in the Event Name field.

### Concrete example
- Some staff enter the event title as "Code4Lib" and some use "Code4Lib 2023".

### Acceptance criteria
- When I review events, there are not duplicate events with/without the year included
- When I try to enter an event name including a year, I am warned and/or prevented from submitting the request

### PR DESCRIPTION
**Pull request by rladdusaw**
Resolves #891

### CODE REVIEW

File: app/javascript/components/travelRequestButton.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        return ""
+      }
+
+      return title.value.match(/\d{2,4}/)[0]
+    }
+  }
+}
+</script>
+
+<style scoped>
```
Comment (ishasinha1) [inline app/javascript/components/travelRequestButton.vue:61]: Do we need this?

File: app/views/travel_requests/_event_date_modal.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="modal-header">
+  <button type="button" class="close" data-dismiss="modal">x</button>
+  <h4 class="modal-title" id="eventDateModal">Possible Date Detected</h4>
+</div>
+<div class="modal-body">
+
```
Comment (ishasinha1) [inline app/views/travel_requests/_event_date_modal.erb:6]: We thought the modal looked nice!

File: app/views/travel_requests/_form.html.erb [hunk 3]
```diff
@@ -12,14 +12,12 @@
         # wanted to see the validations show up on the screen, but did not want to put any effort into the form %>
     <%= form.fields_for :event_requests do |ff| %>
       <%= ff.hidden_field :id %>
-        <input-autocomplete required
-          id="travel_request_event_requests_attributes_0_recurring_event_id"
-          name="travel_request[event_requests_attributes][0][recurring_event_id]"
-          label="Event Name"
-          default-value="<%= request_change_set.event_name %>"
+        <event-title-input-wrapper
+          <% unless request_change_set.event_name.empty? %>
```
Comment (ishasinha1) [inline app/views/travel_requests/_form.html.erb:16]:
```suggestion
          <% if request_change_set.event_name.present? %>
```

Would this be clearer without the double negative?

File: app/views/travel_requests/_form.html.erb [hunk 4]
```diff
@@ -91,7 +89,7 @@
   </grid-item>
   <grid-item columns="lg-3 sm-12 auto">
     <div class="actions">
-      <input-button type="submit" variation="solid" block>Submit Request</input-button>
+      <travel-request-button event-title="<%= request_change_set.event_name %>"></travel-request-button>
```
Comment (ishasinha1) [inline app/views/travel_requests/_form.html.erb:92]: I think this is really cool!

File: spec/features/new_travel_request_spec.rb [hunk 5]
```diff
@@ -90,4 +90,45 @@
...
+    expect(page).to have_content "New Travel Request"
+    click_on "Submit Request"
+
+    expect(page).to have_content "Possible Date Detected"
+    click_on "Continue"
+
+    expect(page).to have_content "Geometry 360 2019, A Place To Be (10/01/2019 to 10/03/2019)"
+    expect(page).to have_content "Airfare 2 20.00 40.00\nTotal: 40.00"
+    expect(page).to have_content "Pending"
+    expect(page).to have_content "I like triangles"
```
Comment (ishasinha1) [inline spec/features/new_travel_request_spec.rb:128]: 🔺

Comment (ishasinha1) [reply spec/features/new_travel_request_spec.rb:128]: Do you actually like triangles?

**Review by ishasinha1 — APPROVED** [review]
Thank you, @rladdusaw! Jane and I noticed a few questions but nothing is a blocker. 🔺

File: app/javascript/components/travelRequestButton.vue (hunk 1 above)
Comment (rladdusaw) [reply app/javascript/components/travelRequestButton.vue:61]: It was included in the other components travelRequestForm and travelRequestDatePicker even when empty so I left it to be consistent.

File: app/views/travel_requests/_event_date_modal.erb (hunk 2 above)
Comment (rladdusaw) [reply app/views/travel_requests/_event_date_modal.erb:6]: Thanks!

File: spec/features/new_travel_request_spec.rb (hunk 5 above)
Comment (rladdusaw) [reply spec/features/new_travel_request_spec.rb:128]: I'm a bit of a math nerd... so yes😆

**Pull request merged**